video/out/vo_gpu_next.c

```diff
@@ -1650,7 +1650,8 @@ static void cache_save_obj(void *p, pl_cache_obj obj)
     }
 
     // Don't save if already exists
-    if (!stat(filepath, &(struct stat){0})) {
+    struct stat st;
+    if (!stat(filepath, &st) && st.st_size > 0) {
```

Comment:
since we're doing it anyway isn't it a good idea to compare the sizes?

Reply:
Good idea.